src/Storages/ObjectStorage/StorageObjectStorageSource.cpp

```diff
@@ -548,6 +548,14 @@ StorageObjectStorageSource::ReaderHolder StorageObjectStorageSource::createReade
         return std::make_shared<ExtractColumnsTransform>(header, read_from_format_info.requested_columns);
     });
 
+    if (configuration->hasDataTransformer(object_info))
```

Comment:
1. how about putting it before `getSchemaTransformer`
2. how about renaming it to `hasPositionalDeletesTransformer`?